site/_posts/2020-02-13-Embedded-Ansible-Part-3-Ansible-In-An-Automation-Workflow.md

```diff
@@ -0,0 +1,270 @@
+ok: [localhost] => {
+    "msg": "Result:True"
+}   
+```
+
+##### get_attribute
+The `get_attribute` function allows the playbook to read an attribute from an object. This example shows the playbook reading the value of the `rhel_subscription_pool` attribute from the configuration domain instance:
+
+```
+- name: Get the "rhel_subscription_pool" attribute from a configuration domain instance (get_attribute)
+  manageiq_automate:
+    workspace: "{{ workspace }}"
+    get_attribute:
+      object: "/Configuration/OpenShift/Parameters/default"
+      attribute: "rhel_subscription_pool"
+  register: rhel_subscription_pool
+- debug: msg="Result:{{ rhel_subscription_pool.value }}"
+```
+
+The output is as follows:
+
+```
+TASK [Get the "rhel_subscription_pool" attribute from a configuration domain instance (get_attribute)] ***
+ok: [localhost]
+
+TASK [debug] *******************************************************************
+ok: [localhost] => {
+    "msg": "Result:Red Hat OpenShift Container Platform, Premium*"
+}
+```
+
+The previous example read an attribute statically defined in a configuration domain instance. This next example uses the same function to read a more dynamic attribute that has been saved as `$evm.root[‘ipam_password’]` by a prior Ruby task in the workflow:
+
+```
+- name: Get the value of the "ipam_passwd" attribute defined by a prior Ruby method (get_attribute)
+  manageiq_automate:
+    workspace: "{{ workspace }}"
+    get_attribute:
+      object: root
+      attribute: "ipam_passwd"
+  register: ipam_password
+- debug: msg="Result:{{ ipam_password.value }}"
+```
+
+##### get_decrypted_attribute
+If an attribute of type “password” has been stored in an instance then it can be accessed and decrypted using the `get_decrypted_attribute` function, for example:
+
+```
+- name: Get the "ldap_user_password" encrypted attribute from a configuration domain instance (get_decrypted_attribute)
+  manageiq_automate:
+    workspace: "{{ workspace }}"
+    get_decrypted_attribute:
+      object: "/Configuration/OpenShift/Parameters/default"
+      attribute: "ldap_user_password"
+  register: ldap_user_password
+- debug: msg="Result:{{ ldap_user_password.value }}"
+```
+
+The output is as follows:
+
+```
+TASK [Get the "ldap_user_password" encrypted attribute from a configuration domain instance (get_decrypted_attribute)] ***
+ok: [localhost]
+
+TASK [debug] *******************************************************************
+ok: [localhost] => {
+    "msg": "Result:{u'attribute': u'ldap_user_password', u'object': u'/Configuration/OpenShift/Parameters/default', u'value': u'secret_ldap_password'}"
+}
+```
+
+Attributes can also be written back into workspace objects (usually the root object) using the `set_attribute`, `set_attributes` and `set_encrypted_attriibute` functions, as follows. Attributes written in this way are readable by a subsequent Ruby method in the workflow as `$evm.root[‘<attribute_name>’]`.
```

Comment:
`set_encrypted_attriibute` - is that a typo or is the function name actually spelled with an extra `i`?

Reply:
Yarps thanks